src/autolabel/data_loaders/validation.py

```diff
@@ -0,0 +1,259 @@
+        column_name_lists += matches
+    return column_name_lists
+
+
+def data_validation_and_schema_check(schema: Dict, validation_task: TaskTypeValidate):
+    """Validate data format and datatype
+
+    Args:
+        schema (Dict): input schema
+        validation_task (TaskTypeValidate): validation task
+
+    Raises:
+        e: Validation error if the inputs are not string
+        e: Validation error if validation_task fails
+
+    Returns:
+        DataValidation: Pydantic Model for validation
+    """
+    Model = create_model("Model", **schema)
+
+    class DataValidation(BaseModel):
+        """Data Validation"""
+
+        @classmethod
+        def validate(cls, data):
+            """Valdiate data types"""
+            model = Model(**data)
+            try:
+                cls(**model.dict())
+            except ValidationError as e:
+                raise e
+
+        @root_validator(pre=True, allow_reuse=True)
+        def check_fields(cls, values):
+            """Validate data format"""
+            try:
+                label_column_value = values[validation_task.label_column]
+                validation_task.validate(label_column_value)
+            except ValidationError as e:
+                raise e
+
+    return DataValidation
+
+
+class TaskDataValidation:
+    """Task Validation"""
+
+    def __init__(
+        self,
+        task_type: str,
+        label_column: str,
+        labels_list: Optional[List],
+        example_template: str,
+    ):
+        """Task Validation
+
+        Args:
+            task_type (str): the type of task, classification, named_entity_recognition, etc.., "config/task_type"
+            label_column (str): the label column as specified in config, "config/dataset/label_column"
+            labels_list (Optional[List]): list of valid labels provided in config "config/prompt/labels"
+            example_template (str): example template from config "config/prompt/example_template"
+        """
+        self.__expected_columns = fetch_expected_columns(example_template)
+        self.__schema = {col: (StrictStr, ...) for col in self.__expected_columns}
+
+        # Initialize Validation task and add attributes
+        self.__validation_task = DataValidationTasks.__dict__[task_type](
+            label_column=label_column, labels_set=set(labels_list)
+        )
+
+    @property
+    def expected_columns(self) -> List:
+        """Fetch expected columns"""
+        return self.__expected_columns
+
+    @property
+    def schema(self) -> Dict:
+        """Fecth Schema"""
+        return self.__schema
+
+    @property
+    def validation_task(
+        self,
+    ) -> TaskTypeValidate:
+        """Fetch validation task"""
+        return self.__validation_task
+
+    def validate(self, data: List[dict]) -> List[Dict]:
+        """Validate Data"""
+        data_validation = data_validation_and_schema_check(
+            self.schema,
+            self.validation_task,
+        )
```

Comment:
we can create this data validator in the constructor?

additionally, please bring the method `data_validation_and_schema_check` in the class